share/eds/store.go

```diff
@@ -312,13 +333,21 @@ func (s *Store) GetCAR(ctx context.Context, root share.DataHash) (io.Reader, err
 	return r, err
 }
 
-func (s *Store) getCAR(ctx context.Context, root share.DataHash) (io.Reader, error) {
-	key := root.String()
-	accessor, err := s.getCachedAccessor(ctx, shard.KeyFromString(key))
+func (s *Store) getCAR(ctx context.Context, root share.DataHash) (io.ReadCloser, error) {
+	key := shard.KeyFromString(root.String())
+	accessor, err := s.cache.Get(key)
+	if err == nil {
+		return newReadCloser(accessor), nil
+	}
+	// If the accessor is not found in the cache, create a new one from dagstore. We don't put accessor
+	// to the cache here, because getCAR is used by shrex.eds. There is a lower probability, compared
+	// to other cache put triggers, that the same block to be requested again.
```

Comment:
```suggestion
	// to the cache put trigger, that the same block will be requested again soon.
```

Reply:
There are could be many triggers, so I've applied it partially